File: app/src/main/java/com/example/jobsearchapp/DatabaseHelper.java
package com.example.jobsearchapp;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DatabaseHelper extends SQLiteOpenHelper {

    private static final String DATABASE_NAME = "JobSearch.db";
    private static final int DATABASE_VERSION = 1;

    public DatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {

        String createUserTable =
                "CREATE TABLE users (" +
                        "id INTEGER PRIMARY KEY AUTOINCREMENT," +
                        "name TEXT," +
                        "email TEXT UNIQUE," +
                        "password TEXT," +
                        "role TEXT)";

        db.execSQL(createUserTable);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

        db.execSQL("DROP TABLE IF EXISTS users");
        onCreate(db);
    }

    // Đăng ký tài khoản
    public boolean insertUser(String name,
                              String email,
                              String password,
                              String role) {

        SQLiteDatabase db = this.getWritableDatabase();

        ContentValues values = new ContentValues();
        values.put("name", name);
        values.put("email", email);
        values.put("password", password);
        values.put("role", role);

        long result = db.insert("users", null, values);

        return result != -1;
    }

    // Kiểm tra email đã tồn tại chưa
    public boolean checkEmail(String email) {

        SQLiteDatabase db = this.getReadableDatabase();

        Cursor cursor = db.rawQuery(
                "SELECT * FROM users WHERE email=?",
                new String[]{email}
        );

        boolean exists = cursor.getCount() > 0;

        cursor.close();

        return exists;
    }

    // Đăng nhập
    public boolean loginUser(String email, String password) {

        SQLiteDatabase db = this.getReadableDatabase();

        Cursor cursor = db.rawQuery(
                "SELECT * FROM users WHERE email=? AND password=?",
                new String[]{email, password}
        );

        boolean success = cursor.getCount() > 0;

        cursor.close();

        return success;
    }
}

File: app/src/main/java/database/DatabaseHelper.kt
package com.example.jobsearchapp.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DatabaseHelper(context: Context) : SQLiteOpenHelper(context, "JobSearch.db", null, 1) {

    override fun onCreate(db: SQLiteDatabase) {
        // 1. Bảng User (Tài khoản)
        val createTableUser = """
            CREATE TABLE User (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                username TEXT,
                email TEXT,
                matKhau TEXT,
                vaiTro TEXT
            )
        """.trimIndent()

        // 2. Bảng Job (Công việc)
        val createTableJob = """
            CREATE TABLE Job (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                tieuDe TEXT,
                tenCongTy TEXT,
                moTa TEXT,
                luong REAL,
                diaDiem TEXT
            )
        """.trimIndent()

        // 3. Bảng Applicant (Hồ sơ ứng viên ứng tuyển)
        val createTableApplicant = """
            CREATE TABLE Applicant (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                jobId INTEGER,
                tenUngVien TEXT,
                emailUngVien TEXT,
                sdtUngVien TEXT,
                trangThai TEXT,
                FOREIGN KEY(jobId) REFERENCES Job(id)
            )
        """.trimIndent()

        db.execSQL(createTableUser)
        db.execSQL(createTableJob)
        db.execSQL(createTableApplicant)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS User")
        db.execSQL("DROP TABLE IF EXISTS Job")
        db.execSQL("DROP TABLE IF EXISTS Applicant")
        onCreate(db)
    }
}